### PR TITLE
Remove warning of 'incompatible type' when compiling

### PR DESCRIPTION
**Pull request by kuien**
Files changed:
    modified:   src/backend/cdb/motion/ic_common.c
    modified:   src/backend/executor/spi.c
    modified:   src/backend/nodes/outfuncs.c
    modified:   src/backend/optimizer/path/costsize.c
    modified:   src/backend/storage/file/compress_zlib.c

Note: The warning indicates a potential error in
    function _outScanInfo() of outfuncs.c.

### CODE REVIEW

File: src/backend/cdb/motion/ic_common.c [hunk 1]
```diff
@@ -271,7 +271,7 @@ InitMotionLayerIPC(void)
 	/*activated = false;*/
 	savedSeqServerFd = -1;
 
-	InitMotionUDPIFC(&UDP_listenerFd, &Gp_listener_port);
+	InitMotionUDPIFC(&UDP_listenerFd, (uint16 *) &Gp_listener_port);
```
Comment (hlinnaka) [inline src/backend/cdb/motion/ic_common.c]: This won't do what you want on a big-endian system. I'd suggest just changing the argument type of InitMotionUDPIFC(). Or if you want a local fix, something like:

uint16 port;

InitMotionUDPIFC(&UDP_listenerFd, &port);
Gp_listener_port = port;

File: src/backend/nodes/outfuncs.c [hunk 2]
```diff
@@ -404,7 +404,7 @@ _outScanInfo(StringInfo str, Scan *node)
 		 * subquery, but this is serialized elsewhere in outSubqueryScan, no
 		 * need to duplicate it here
 		 */
```
Comment (hlinnaka) [inline src/backend/nodes/outfuncs.c]: This is dead code, let's remove it altogether.

print_variable_fields is initialized to true, and only set to false in outfuncs_workfile_mgr_init() function. But outfuncs_workfile_mgr_init() is never called. So we can remove all of: print_variable_fields, outfuncs_workfile_mgr_init(), outfuncs_workfile_mgr_end(), and all the if() tests depending on print_variable_fields, including this one.

File: src/backend/cdb/motion/ic_common.c (hunk 1 above)
Comment (kuien) [inline src/backend/cdb/motion/ic_common.c]: Cool, I prefer the local fix, otherwise I may bring into diffs more than 5 places.

Comment (kuien) [inline src/backend/cdb/motion/ic_common.c]: Cool, I prefer the local fix :)

File: src/backend/nodes/outfuncs.c (hunk 2 above)
Comment (kuien) [inline src/backend/nodes/outfuncs.c]: OK, I'll do this in an separated PR.

And, @xiong-gang what are your thoughts?  Last time, if I remember correctly, you mentioned these functions are still useful in dispatching.

**Comment by HaozhouWang** [comment]
LGTM

**Comment by kuien** [comment]
merged, thanks. 

The dead code in outfuncs.c will be remove in a separated PR, and similar code blocks in outfast.c would be investigated further.

**Pull request merged**